File: 0485-max-consecutive-ones/0485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        l = 0
        ans = 0

        for r in range(len(nums)):
            if nums[r] == 1:
                ans = max(ans, r - l + 1)
            else:
                l = r + 1
        
        return ans
        

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    let result = 0;
    let counter = 0;
    
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] === 0) {
            result = Math.max(counter, result);
            counter = 0;
        } else {
            counter++;
        }
    }
    
    result = Math.max(counter, result);
    return result;
};